File: python/Challenge 3/timeConversion.py
def timeConversion(s):
    # Write your code here

    hour = s[:2]
    min_secs = s[2:len(s) - 2]
    aux = s[len(s) - 2: len(s)]

    if aux == "PM" and int(hour) == 12:
        return hour + min_secs
    elif aux == "AM" and int(hour) == 12:
        return "00" + min_secs
    elif aux == "AM":
        return hour + min_secs
    elif aux == "PM":
        return f"{int(hour) + 12}" + min_secs
    

s = "12:40:22AM"
print(timeConversion(s))

File: javaScript/Challenge 3/timeConversion.js
function timeConversion(s) {
    // Write your code here

    let convertedDate = "";

    let hour = s.slice(0, 2);
    const minSecs = s.slice(2, s.length - 2);
    const aux = s.slice(s.length - 2, s.length);

    if (parseInt(hour) === 12 && aux === "PM") convertedDate = hour + minSecs;
    else if (parseInt(hour) === 12 && aux === "AM") convertedDate = "00" + minSecs;
    else if (aux === "PM") convertedDate = `${parseInt(hour) + 12}` + minSecs;
    else if (aux === "AM") convertedDate = hour + minSecs;

    return convertedDate;
}


const s = "12:40:22AM";

console.log(timeConversion(s));